atomic_reactor/plugins/post_flatpak_create_oci.py

```diff
@@ -75,112 +41,52 @@ def __init__(self, workflow):
-        :param tag: str, tag to apply to name in the internal storage
-        """
-        skopeo_copy_dst = '{}:{}:{}'.format(self.workflow.storage_transport, name, tag)
-        cmd = ['skopeo', 'copy', 'oci:{}'.format(oci_path), skopeo_copy_dst]
-
-        self.log.info("Copying built image to internal container image storage as %s:%s", name, tag)
-        try:
-            retries.run_cmd(cmd)
-        except subprocess.CalledProcessError as e:
-            self.log.error("image copy failed with output:\n%s", e.output)
-            raise
-
-    def run(self):
-        if not is_flatpak_build(self.workflow):
-            self.log.info('not flatpak build, skipping plugin')
-            return
-
-        source = get_flatpak_source_info(self.workflow)
-        if source is None:
-            raise RuntimeError("flatpak_create_dockerfile must be run before flatpak_create_oci")
-
-        self.builder = FlatpakBuilder(source, self.workflow.source.workdir,
+    def build_flatpak_image(self, source, build_dir: BuildDir) -> Dict[str, Union[str, int]]:
+        self.builder = FlatpakBuilder(source, build_dir.path,
                                       'var/tmp/flatpak-build',
                                       parse_manifest=parse_rpm_output,
                                       flatpak_metadata=self.flatpak_metadata)
 
-        df_labels = df_parser(self.workflow.df_path, workflow=self.workflow).labels
-        self.builder.add_labels(df_labels)
+        df_labels = build_dir.dockerfile_with_parent_env(
+            self.workflow.imageutil.base_image_inspect()
+        ).labels
 
-        tarred_filesystem, manifest = self._export_filesystem()
-        self.log.info('filesystem tarfile written to %s', tarred_filesystem)
-        self.log.info('manifest written to %s', manifest)
+        self.builder.add_labels(df_labels)
 
-        image_components = self.builder.get_components(manifest)
-        self.workflow.data.image_components = image_components
+        tmp_dir = tempfile.mkdtemp(dir=build_dir.path)
 
-        ref_name, outfile, tarred_outfile = self.builder.build_container(tarred_filesystem)
+        image_filesystem = self.workflow.imageutil.extract_filesystem_layer(
+            str(build_dir.exported_squashed_image), str(tmp_dir))
 
-        # OSBS2 TBD
-        self.log.info('Marking filesystem image "%s" for removal', self.workflow.data.image_id)
+        filesystem_path = os.path.join(tmp_dir, image_filesystem)
 
-        image_id = self._get_oci_image_id(outfile)
-        self.log.info('New OCI image ID is %s', image_id)
-        # OSBS2 TBD
-        self.workflow.data.image_id = image_id
+        with open(filesystem_path, 'rb') as f:
+            # this part is 'not ideal' but this function seems to be a prerequisite
+            # for building flatpak image since it does the setup for it
+            flatpak_filesystem, _ = self.builder._export_from_stream(f)
```

Comment:
This part is really cumbersome since we're calling an internal function that is reprocessing the filesystem but without this call, the `build_container` call fails.

Reply:
Turns out we already used the private function in OSBS 1, so this is probably the best we can do :disappointed: 

https://github.com/containerbuildsystem/atomic-reactor/blob/d7ec9ef5ba2d230aa70d12905f37952a1d2f1414/atomic_reactor/plugins/prepub_flatpak_create_oci.py#L81

Reply:
cool